### PR TITLE
Docs cleanups, clarifications, and fixes

### PR DESCRIPTION
**Pull request by jrudolph**
I'm still working on the bootstrap details page which I find a bit confusing because it is quite redundant.

### CODE REVIEW

**Comment by chbatey** [comment]
It was copy pastered from the original one page doc so probably a lot can be cut out

File: docs/src/main/paradox/akka-management.md [hunk 1]
```diff
@@ -147,7 +120,9 @@ base path. For example, when using Akka Cluster Management routes the members in
 
 @@@ note
 
-HTTPS is not enabled by default as additional configuration from the developer is required This module does not provide security by default. It's the developer's choice to add security to this API.
+HTTPS is not enabled by default as additional configuration from the developer is required This module does not provide security by default.
```
Comment (raboof) [inline docs/src/main/paradox/akka-management.md]: missing '.' after 'required'

File: docs/src/main/paradox/akka-management.md [hunk 2]
```diff
@@ -207,35 +182,35 @@ Scala
...
 @@@
 
 ## Stopping Akka Management
 
 In a dynamic environment you might stop instances of Akka Management, for example if you don't want to free up resources
-taken by the HTTP server serving the Management routes. 
+taken by the HTTP server serving the management routes.
 
-You can do so by calling `stop()` on @scaladoc[AkkaManagement](akka.management.http.AkkaManagement). 
+You can do so by calling `stop()` on @scaladoc[AkkaManagement](akka.management.http.AkkaManagement).
```
Comment (raboof) [inline docs/src/main/paradox/akka-management.md]: No unidoc here yet?

File: docs/src/main/paradox/bootstrap/index.md [hunk 3]
```diff
@@ -57,55 +59,69 @@ Gradle
...
+  to be using the Bootstrap mechanism.
 @@@
 
-## Bootstrap recipes 
+## Bootstrap Recipes
+
+To see how to configure and use bootstrap in various environments such as Kubernetes, see @ref[recipes](recipes.md).
 
-To see how to use bootstrap in various environments such as Kubernetes see @ref[recipes](recipes.md)
+@@@ index
```
Comment (raboof) [inline docs/src/main/paradox/bootstrap/index.md]: :+1:

File: docs/src/main/paradox/akka-management.md [hunk 4]
```diff
@@ -1,43 +1,16 @@
...
-
-Management extensions whose nature is "active" (as in, they "do something proactively") should not be
-started automatically, but instead be started manually by the user. One example of that is the Cluster
-Bootstrap, which on one hand does contributes routes to Akka Management, however the bootstraping process
-does not start unless `ClusterBootstrap().start()` is invoked, which allows the user to decide when exactly
-it is ready and wants to start joining an existing cluster.
-
-![project structure](images/structure.png)
+Akka Management is the core module of the management utilities that provides a central HTTP endpoint for Akka
+management extensions.
```
Comment (helena) [inline docs/src/main/paradox/akka-management.md]: I might use ```which provides``` versus that.

File: docs/src/main/paradox/akka-management.md [hunk 5]
```diff
@@ -1,43 +1,16 @@
...
 
 ## Dependencies
 
 Akka management requires Akka 2.5 or later.
 
 The main Akka Management dependency is called `akka-management`. By itself however it does not provide any capabilities,
 and you have to combine it with the management extension libraries that you want to make use of (e.g. cluster http management,
-or cluster bootstrap). This design choice was made to make users able to include only the minimal set of features that they 
+or cluster bootstrap). This design choice was made to make users able to include only the minimal set of features that they
```
Comment (helena) [inline docs/src/main/paradox/akka-management.md]:
```suggestion
or cluster bootstrap). This design choice enables users to include only the minimal set of features they
```

File: docs/src/main/paradox/akka-management.md [hunk 6]
```diff
@@ -63,8 +36,8 @@ Java
 :   ```java
     AkkaManagement.get(system).start();
     ```
-    
-This is in order to allow users to prepare anything else they might want to prepare or start before exposing routes for 
+
+This is in order to allow users to prepare anything else they might want to prepare or start before exposing routes for
```
Comment (helena) [inline docs/src/main/paradox/akka-management.md]:
```suggestion
This allows users to prepare anything further before exposing routes for
```

File: docs/src/main/paradox/akka-management.md (hunk 1 above)
Comment (helena) [inline docs/src/main/paradox/akka-management.md]:
```suggestion
HTTPS is not enabled by default, as additional configuration from the developer is required. This module does not provide security by default.
```

File: docs/src/main/paradox/akka-management.md [hunk 7]
```diff
@@ -147,7 +120,9 @@ base path. For example, when using Akka Cluster Management routes the members in
 
 @@@ note
 
-HTTPS is not enabled by default as additional configuration from the developer is required This module does not provide security by default. It's the developer's choice to add security to this API.
+HTTPS is not enabled by default as additional configuration from the developer is required This module does not provide security by default.
+It's the developer's choice to add security to this API. It is generally advisable not to expose management endpoints
```
Comment (helena) [inline docs/src/main/paradox/akka-management.md]:
```suggestion
It is the developer's choice to add security to this API, and when. If enabled, it is generally advisable not to expose management endpoints
```

File: docs/src/main/paradox/akka-management.md [hunk 8]
```diff
@@ -147,7 +120,9 @@ base path. For example, when using Akka Cluster Management routes the members in
...
+on the internet.
```
Comment (helena) [inline docs/src/main/paradox/akka-management.md]: Sounds very arpanet, might we say something indicating publicly or not on a private network vs this word?

File: docs/src/main/paradox/akka-management.md [hunk 9]
```diff
@@ -255,3 +230,26 @@ Java
     httpClusterManagement.stop();
     ```
 
+## Developing Extensions
+
+This project provides a set of management extensions. If you want to write third-party extensions to Akka management here
```
Comment (helena) [inline docs/src/main/paradox/akka-management.md]:
```suggestion
This project provides a set of management extensions. To write third-party extensions to Akka Management, here
```

File: docs/src/main/paradox/akka-management.md [hunk 10]
```diff
@@ -255,3 +230,26 @@ Java
...
+
+The `akka-management` module provides the central HTTP endpoint to which extensions can register themselves.
+
+For example, the "Cluster HTTP Management" module exposes HTTP routes that can be used to monitor,
+and even trigger joining/leaving/downing decisions via HTTP calls to these routes. The routes and
+logic for these are implemented inside the `akka-management-cluster-http`.
+
+An extension can contribute to the exposed HTTP routes by appending to the `akka.management.http.route-providers` list in
+its own `reference.conf` (make sure to use `+=` instead of `=`). The core `AkkaManagement` extension
+then collects all the routes and serves them together under the management HTTP server. This is in order
```
Comment (helena) [inline docs/src/main/paradox/akka-management.md]: ```management``` vs ```Management```? I don't know, just bringing it up

Comment (helena) [inline docs/src/main/paradox/akka-management.md]:
```suggestion
then collects all the routes and serves them together under the management HTTP server. This
```

File: docs/src/main/paradox/akka-management.md [hunk 11]
```diff
@@ -255,3 +230,26 @@ Java
...
+to avoid having to start an additional HTTP server for each additional extension, and also, it allows
```
Comment (helena) [inline docs/src/main/paradox/akka-management.md]:
```suggestion
avoids having to start an additional HTTP server for each additional extension, and additionally allows
```

File: docs/src/main/paradox/akka-management.md [hunk 12]
```diff
@@ -255,3 +230,26 @@ Java
...
+easy extension of routes served by including libraries that offer new capabilities (such as health-checks or
+cluster information etc).
```
Comment (helena) [inline docs/src/main/paradox/akka-management.md]: Kind of run-on, maybe break the sentence up.

File: docs/src/main/paradox/akka-management.md [hunk 13]
```diff
@@ -255,3 +230,26 @@ Java
...
+
+Management extensions whose nature is "active" (as in, they "do something proactively") should not be
```
Comment (helena) [inline docs/src/main/paradox/akka-management.md]:
```suggestion
Management extensions whose nature is "active", e.g. they do something proactively, should not be
```

File: docs/src/main/paradox/akka-management.md [hunk 14]
```diff
@@ -255,3 +230,26 @@ Java
...
+started automatically, but should instead be started manually by the user. One example of that is the Cluster
```
Comment (helena) [inline docs/src/main/paradox/akka-management.md]:
```suggestion
started automatically, but rather started manually by the user. One example of that is the Cluster
```

File: docs/src/main/paradox/akka-management.md [hunk 15]
```diff
@@ -255,3 +230,26 @@ Java
...
+Bootstrap, which on one hand does contributes routes to Akka Management, however the bootstrapping process
+does not start unless `ClusterBootstrap().start()` is invoked, which allows the user to decide when exactly
+it is ready and wants to start joining an existing cluster.
```
Comment (helena) [inline docs/src/main/paradox/akka-management.md]: A run-on. Perhaps:
```suggestion
One example of the recommended manual operations is Cluster Bootstrap, which contributes routes to Akka Management. The bootstrapping process does not start unless `ClusterBootstrap().start()` is invoked, allowing the user to decide exactly when it is ready to start joining an existing cluster.
```

File: docs/src/main/paradox/bootstrap/index.md [hunk 16]
```diff
@@ -1,10 +1,12 @@
 <a id="bootstrap"></a>
 # Akka Cluster Bootstrap
 
-Akka Cluster Bootstrap supports safely forming a new cluster from discovered nodes or joining a node to an existing cluster. 
-It is an alternative to configuring  static `seed-nodes`.
+Akka Cluster Bootstrap helps forming (or joining to) a cluster by using @ref:[Akka Discovery](../discovery/index.md)
+to discover peer nodes.  It is an alternative to configuring  static `seed-nodes` in common dynamic deployment environments
```
Comment (helena) [inline docs/src/main/paradox/bootstrap/index.md]:
```suggestion
to discover peer nodes.  It is an alternative to configuring static `seed-nodes` in common dynamic deployment environments
```

File: docs/src/main/paradox/bootstrap/index.md [hunk 17]
```diff
@@ -1,10 +1,12 @@
...
+like on kubernetes or on AWS.
```
Comment (helena) [inline docs/src/main/paradox/bootstrap/index.md]:
```suggestion
such as on Kubernetes or AWS.
```

File: docs/src/main/paradox/bootstrap/index.md [hunk 18]
```diff
@@ -1,10 +1,12 @@
...
 
-Akka Cluster Bootstrap does not require any additional system like etcd/zookeeper/consul to be run along side the Akka cluster in order to discover the seed-nodes.
+It builds on the flexibility of Akka Discovery to make use of a range of different discovery mechanisms depending on the
```
Comment (helena) [inline docs/src/main/paradox/bootstrap/index.md]:
```suggestion
It builds on the flexibility of Akka Discovery, leveraging a range of discovery mechanisms depending on the
```

File: docs/src/main/paradox/bootstrap/index.md [hunk 19]
```diff
@@ -57,55 +59,69 @@ Gradle
...
-As Akka Cluster allows nodes to join to a cluster using a few methods it is the precedence of each method
+As Akka Cluster allows nodes to join to a cluster using a few methods, the precedence of each method
 is strictly defined and is as follows:
 
-- If configured `akka.cluster.seed-nodes` (in your `application.conf`) are non-empty, those nodes will be joined, and bootstrap will NOT execute even if `start()` is called (a warning will be logged though).
-- If an explicit `cluster.join` or `cluster.joinSeedNodes` is invokes before the bootstrap completes that
- joining would take precedence over the bootstrap. *This is however **not** recommended to mix multiple
- joining methods like this.*
+- If `akka.cluster.seed-nodes` (in your `application.conf`) are non-empty, those nodes will be joined, and bootstrap
+  will NOT execute even if `start()` is called (a warning will be logged, though).
```
Comment (helena) [inline docs/src/main/paradox/bootstrap/index.md]:
```suggestion
  will NOT execute even if `start()` is called, however a warning will be logged.
```

File: docs/src/main/paradox/bootstrap/index.md [hunk 20]
```diff
@@ -57,55 +59,69 @@ Gradle
...
+- If an explicit `cluster.join` or `cluster.joinSeedNodes` is invoked before the bootstrap completes that
```
Comment (helena) [inline docs/src/main/paradox/bootstrap/index.md]:
```suggestion
- If an explicit `cluster.join` or `cluster.joinSeedNodes` is invoked before the bootstrap completes, that
```

File: docs/src/main/paradox/bootstrap/kubernetes-api.md [hunk 21]
```diff
@@ -1,24 +1,24 @@
-# Kubernetes API 
+# Kubernetes API
 
 An example project that can be deployed to kubernetes via `minikube` is in `bootstrap-demo/kubernetes-api`.
```
Comment (helena) [inline docs/src/main/paradox/bootstrap/kubernetes-api.md]:
```suggestion
An example project that can be deployed to Kubernetes via `Minikube` is in `bootstrap-demo/kubernetes-api`.
```

File: docs/src/main/paradox/akka-management.md (hunk 1 above)
Comment (ignasi35) [reply docs/src/main/paradox/akka-management.md]: @raboof try out the new `Insert a suggestion` in GH UI. :-)

Comment (ignasi35) [reply docs/src/main/paradox/akka-management.md]:
```suggestion
HTTPS is not enabled by default as additional configuration from the developer is required. This module does not provide security by default.
```

File: docs/src/main/paradox/akka-management.md (hunk 10 above)
Comment (jrudolph) [reply docs/src/main/paradox/akka-management.md]: Yeah, good point. I guess we are a bit sloppy on deciding on consistent terms and following through with them.

Comment (jrudolph) [reply docs/src/main/paradox/akka-management.md]: Added https://github.com/lightbend/paradox/issues/274 to add a term/spell checker to paradox (though, that wouldn't have helped here I guess).

File: docs/src/main/paradox/akka-management.md (hunk 12 above)
Comment (jrudolph) [reply docs/src/main/paradox/akka-management.md]: Yep, reworked the whole section a bit.

File: docs/src/main/paradox/akka-management.md (hunk 2 above)
Comment (jrudolph) [reply docs/src/main/paradox/akka-management.md]: No unidoc in akka-management afaics.

**Comment by jrudolph** [comment]
Thanks for the reviews and sorry for keeping that hanging after my vacation. I fixed a few more small things.

**Comment by jrudolph** [comment]
Created https://github.com/akka/akka-management/issues/388 to deduplicate the bootstrap details page as I don't get to that immediately.

**Pull request merged**